File: 27-system-ai/2-system-design-basics/main.py
# System Design Basics - Python example (Simulated Load Balancer)

class LoadBalancer:
    def __init__(self, servers):
        self.servers = servers
        self.index = 0

    def get_server(self):
        server = self.servers[self.index]
        self.index = (self.index + 1) % len(self.servers)
        return server

if __name__ == "__main__":
    lb = LoadBalancer(['server1', 'server2', 'server3'])
    for _ in range(5):
        print(lb.get_server())

File: 27-system-ai/2-system-design-basics/main.java
// System Design Basics - Java example (Simulated Load Balancer)

class Main {
    static class LoadBalancer {
        private String[] servers;
        private int index = 0;

        public LoadBalancer(String[] servers) {
            this.servers = servers;
        }

        public String getServer() {
            String server = servers[index];
            index = (index + 1) % servers.length;
            return server;
        }
    }

    public static void main(String[] args) {
        LoadBalancer lb = new LoadBalancer(new String[]{"server1", "server2", "server3"});
        for (int i = 0; i < 5; i++) {
            System.out.println(lb.getServer());
        }
    }
}
